pixeltable/catalog/catalog.py

```diff
@@ -2404,6 +2410,16 @@ def _handle_path_collision(
         if if_exists == IfExistsParam.IGNORE:
             return obj
 
+        assert if_exists in (IfExistsParam.REPLACE, IfExistsParam.REPLACE_FORCE)
+
+        # Check for circularity
```

Comment:
Is this applicable beyond create_view()? If not, it would be easier to follow if the logic lives in create_view().

Reply:
I agree, but that would need a larger refactoring, since we need to do the circularity check *after* resolving the name collision but *before* dropping the table that's being replaced, and both those operations happen in `_handle_path_collision()`.

I guess in theory we could safely drop the table first as long as the circularity exception is thrown inside the same transaction, but it feels wrong to rely on that (and it would increase the load on the database, invite additional concurrency conflicts etc., all unnecessarily).